sbol_utilities/component.py

```diff
@@ -60,6 +60,26 @@ def check_roles(component: sbol3.Component) -> bool:
         and has_dna_type(obj) and len(obj.sequences) == 1
 
 
+def by_roles(obj: sbol3.TopLevel, required_role: str) -> bool:
```

Comment:
This can be cleaner if instead of returning a boolean, you return a lambda that returns a boolean. Then you can construct the function and pass the function into filter_top_level rather than passing its arguments into filter_top_level